projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php

```diff
@@ -97,6 +105,110 @@ public static function is_banner_enabled() {
 		return $recommendations_banner_enabled;
 	}
 
+	/**
+	 * Set up actions to monitor for things that trigger a recommendation.
+	 *
+	 * @return false|void
+	 */
+	public static function init_conditional_recommendation_actions() {
+		// Check to make sure that recommendations are enabled.
+		if ( ! self::is_enabled() ) {
+			return false;
+		}
+
+		// Monitor for the publishing of a new post.
+		add_action( 'transition_post_status', array( 'Jetpack_Recommendations', 'publicize_recommendation_post_transition' ), 10, 3 );
```

Comment:
Since we're all in the same class you should be able to simplify things here.

Reply:
I've made some changes here - LMK if you had another thought on how to improve this :)